online/src/app/store/actions/product.actions.ts

```diff
@@ -0,0 +1,79 @@
+import { HttpErrorResponse } from "@angular/common/http";
+import { createAction, props } from "@ngrx/store";
+import { Product } from "src/app/products";
+import { ProductViewModel } from "src/app/products";
+
+export const getProduct = createAction(
+    '[Product] Get Product',
+    props<{ id: number }>()
+)
+
+export const getProductSucces = createAction(
+    '[Product] Get Product Success',
+    props<{ product: Product }>()
+)
+
+export const getProductError = createAction(
+    '[Product] Get Product Error',
+    props<{ response: HttpErrorResponse }>()
+)
+
+export const addProduct = createAction(
+    '[Product] Add Product',
+    props<{ product: Product }>()
+)
+
+export const addProductSuccess = createAction(
+    '[Product] Add Product Success',
+    props<{ product: Product }>()
+)
+
+export const addProductError = createAction(
```

Comment:
i would only make 1 action for errors and have the message as param tbh, such a hassle to maintain many error actions